jerry-core/ecma/base/ecma-errors.c

```diff
@@ -15,202 +15,53 @@
-#endif /* JERRY_MODULE_SYSTEM */
-
-/**
- * Error message, if callback function is not callable
- */
-const char* const ecma_error_callback_is_not_callable = "Callback function is not callable";
-
-/**
- * Error message, if arrayBuffer is detached
- */
-const char* const ecma_error_arraybuffer_is_detached = "ArrayBuffer has been detached";
-
-/**
- * Error message, cannot convert undefined or null to object
- */
-const char* const ecma_error_cannot_convert_to_object = "Cannot convert undefined or null to object";
-
-/**
- * Error message, prototype property of a class is non-configurable
- */
-const char* const ecma_error_class_is_non_configurable = "Prototype property of a class is non-configurable";
-
-/**
- * Error message, argument is not an object
- */
-const char* const ecma_error_argument_is_not_an_object = "Argument is not an object";
-
-/**
- * Error message length of 'ecma_error_argument_is_not_an_object'
- */
-const int ecma_error_argument_is_not_an_object_length = 25;
-
-/**
- * Error message, argument is not a Proxy object
- */
-const char* const ecma_error_argument_is_not_a_proxy = "Argument is not a Proxy object";
-
-/**
- * Error message, target is not a constructor
- */
-const char* const ecma_error_target_is_not_a_constructor = "Target is not a constructor";
-
-/**
- * Error message, argument is not an regexp
- */
-const char* const ecma_error_argument_is_not_an_regexp = "Argument 'this' is not a RegExp object";
-
-/**
- * Error message, invalid array length
- */
-const char* const ecma_error_invalid_array_length = "Invalid Array length";
-
-/**
- * Error message, local variable is redeclared
- */
-const char* const ecma_error_local_variable_is_redeclared = "Local variable is redeclared";
-
-/**
- * Error message, expected a function
- */
-const char* const ecma_error_expected_a_function = "Expected a function";
-
-#if JERRY_ESNEXT
+#include "jrt.h"
 
 /**
- * Error message, class constructor invoked without new keyword
- */
-const char* const ecma_error_class_constructor_new = "Class constructor cannot be invoked without 'new'";
-
-/**
- * Error message, variables declared by let/const must be initialized before reading their value
+ * Get specified ecma error as zero-terminated string
+ *
+ * @return pointer to zero-terminated ecma error
  */
-const char* const ecma_error_let_const_not_initialized = ("Variables declared by let/const must be"
-                                                          " initialized before reading their value");
-
-#endif /* JERRY_ESNEXT */
-
+const lit_utf8_byte_t *
+get_ecma_error_utf8 (uint32_t id) /**< ecma error id */
+{
+  JERRY_ASSERT (id != ECMA_IS_VALID_CONSTRUCTOR);
+#if JERRY_ERROR_MESSAGES
+  static const lit_utf8_byte_t *const ecma_error[] JERRY_ATTR_CONST_DATA = {
+/** @cond doxygen_suppress */
+#define ECMA_ERROR_DEF(id, utf8_string) (const lit_utf8_byte_t *) utf8_string,
+#include "ecma-err-msg.inc.h"
+#undef ECMA_ERROR_DEF
+    /** @endcond */
+  };
```

Comment:
* Is there a reason to have this array inside the method?
* Is there any reason not to only include the error messages inc.h once? Like for example just before the methods we could create a struct which have both the strings and size:
```c
#if JERRY_ERROR_MESSAGES

static const struct {
  list_utf8_byte_t *const text;
  uint32_t size;
} ecma_error_messages[] JERRY_ATTR_CONST_DATA = {
/** @cond doxygen_suppress */
#define ECMA_ERROR_DEF(id, utf8_string) \
  { (const lit_utf8_byte_t *) utf8_string, sizeof (utf8_string) - 1 },

#include "ecma-err-msg.inc.h"

#undef ECMA_ERROR_DEF
/** @endcond */
};
#endif

```